cpp/unreal_plugins/SimulationController/Source/SimulationController/WheeledVehicleAgent.cpp

```diff
@@ -0,0 +1,310 @@
+    }
+    else {
+        SP_ASSERT(false);
+    }
+    FActorSpawnParameters actor_spawn_params;
+    actor_spawn_params.Name = Unreal::toFName(Config::get<std::string>("SIMULATION_CONTROLLER.WHEELED_VEHICLE_AGENT.WHEELED_VEHICLE_ACTOR_NAME"));
+    actor_spawn_params.SpawnCollisionHandlingOverride = ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
+    wheeled_vehicle_pawn_ = world->SpawnActor<AVehiclePawn>(spawn_location, spawn_rotation, actor_spawn_params);
+    SP_ASSERT(wheeled_vehicle_pawn_);
+
+    auto observation_components = Config::get<std::vector<std::string>>("SIMULATION_CONTROLLER.WHEELED_VEHICLE_AGENT.OBSERVATION_COMPONENTS");
+
+    if (Std::contains(observation_components, "camera")) {
+        camera_sensor_ = std::make_unique<CameraSensor>(
+            wheeled_vehicle_pawn_->camera_component_,
+            Config::get<std::vector<std::string>>("SIMULATION_CONTROLLER.WHEELED_VEHICLE_AGENT.CAMERA.RENDER_PASSES"),
+            Config::get<unsigned int>("SIMULATION_CONTROLLER.WHEELED_VEHICLE_AGENT.CAMERA.IMAGE_WIDTH"),
+            Config::get<unsigned int>("SIMULATION_CONTROLLER.WHEELED_VEHICLE_AGENT.CAMERA.IMAGE_HEIGHT"),
+            Config::get<float>("SIMULATION_CONTROLLER.WHEELED_VEHICLE_AGENT.CAMERA.FOV"));
+        SP_ASSERT(camera_sensor_);
+    }
+
+    if (Std::contains(observation_components, "imu")) {
+        imu_sensor_ = std::make_unique<ImuSensor>(wheeled_vehicle_pawn_->imu_component_);
+        SP_ASSERT(imu_sensor_);
+    }
+
+    if (Std::contains(observation_components, "sonar")) {
+        sonar_sensor_ = std::make_unique<SonarSensor>(wheeled_vehicle_pawn_->sonar_component_);
+        SP_ASSERT(sonar_sensor_);
+    }
+}
+
+WheeledVehicleAgent::~WheeledVehicleAgent()
+{
+    SP_LOG_CURRENT_FUNCTION();
+
+    auto observation_components = Config::get<std::vector<std::string>>("SIMULATION_CONTROLLER.WHEELED_VEHICLE_AGENT.OBSERVATION_COMPONENTS");
+
+    if (Std::contains(observation_components, "sonar")) {
+        SP_ASSERT(sonar_sensor_);
+        sonar_sensor_ = nullptr;
+    }
+
+    if (Std::contains(observation_components, "imu")) {
+        SP_ASSERT(imu_sensor_);
+        imu_sensor_ = nullptr;
+    }
+
+    if (Std::contains(observation_components, "camera")) {
+        SP_ASSERT(camera_sensor_);
+        camera_sensor_ = nullptr;
+    }
+
+    SP_ASSERT(wheeled_vehicle_pawn_);
+    wheeled_vehicle_pawn_->Destroy();
+    wheeled_vehicle_pawn_ = nullptr;
+}
+
+void WheeledVehicleAgent::findObjectReferences(UWorld* world) {}
+
+void WheeledVehicleAgent::cleanUpObjectReferences() {}
+
+std::map<std::string, ArrayDesc> WheeledVehicleAgent::getActionSpace() const
+{
+    std::map<std::string, ArrayDesc> action_space;
+    auto action_components = Config::get<std::vector<std::string>>("SIMULATION_CONTROLLER.WHEELED_VEHICLE_AGENT.ACTION_COMPONENTS");
+
+    if (Std::contains(action_components, "apply_wheel_torques")) {
```

Comment:
Factor these actions in a cleaner way. The actions should be: `set_drive_torque`, `set_brake_torque`. We aren't using these other actions for anything.

Reply:
Agreed, updated this.